src/MySqlConnector/Protocol/Serialization/ByteBufferWriter.cs

```diff
@@ -226,7 +237,8 @@ private void Reallocate(int additional = 0)
 		m_output = new(m_buffer, usedLength, m_buffer.Length - usedLength);
 	}
 
-	private Encoder? m_encoder;
+	private static Encoder m_utf8Encoder = Encoding.UTF8.GetEncoder();
```

Comment:
This type isn't thread-safe and can't be shared across multiple instances of `ByteBufferWriter`. In particular, "An [Encoder](https://learn.microsoft.com/en-us/dotnet/api/system.text.encoder?view=net-7.0) object maintains state information between successive calls to GetBytes or [Convert](https://learn.microsoft.com/en-us/dotnet/api/system.text.decoder.convert?view=net-7.0) methods so that it can correctly encode character sequences that span blocks."

Data would very likely be corrupted if `m_utf8Encoder` was used simultaneously by multiple threads.